app/src/main/java/com/elmeradrianv/shesafe/SignUpActivity.java

```diff
@@ -0,0 +1,177 @@
+            String passwordConfirm = etPasswordConfirm.getText().toString();
+            if (username.isEmpty() && firstName.isEmpty() && lastName.isEmpty() && email.isEmpty() && personalDescription.isEmpty() && password.isEmpty() && passwordConfirm.isEmpty()) {
+                Toast.makeText(this, "Please fill all the fields", Toast.LENGTH_SHORT).show();
+            } else {
+                if (password.equals(passwordConfirm)) {
+                    signupNewUser(username, firstName, lastName, email, personalDescription, password);
+                } else {
+                    Toast.makeText(this, "Password doesn't matched", Toast.LENGTH_SHORT).show();
+                }
+            }
+
+        });
+    }
+
+
+    private void signupNewUser(String username, String firstName, String lastName, String email, String personalDescription, String password) {
+        ParseUser user = new ParseUser();
+        user.setUsername(username);
+        user.setEmail(email);
+        user.setPassword(password);
+        user.put(User.PERSONAL_DESCRIPTION_KEY, personalDescription);
+        user.put(User.FIRST_NAME_KEY, firstName);
+        user.put(User.LAST_NAME_KEY, lastName);
+        user.signUpInBackground(e2 -> {
+            if (e2 != null) {
+                Toast.makeText(this, "Couldn't sign up", Toast.LENGTH_SHORT).show();
+                Log.e(TAG, "signupNewUser: Signup error", e2);
+                return;
+            }
+            if (PERSONALIZED_PHOTO_PICKED) {
+                user.put(User.PROFILE_PHOTO_KEY, profilePhoto);
+                user.saveInBackground((SaveCallback) e3 -> {
+                    if (e2 != null) {
+                        Toast.makeText(this, "Couldn't sign up", Toast.LENGTH_SHORT).show();
+                        Log.e(TAG, "signupNewUser: Signup error", e2);
+                        return;
+                    }
+                });
+                PERSONALIZED_PHOTO_PICKED = false;
+                profilePhoto = null;
+            }
+            Toast.makeText(this, "Success!", Toast.LENGTH_SHORT).show();
+        });
+    }
+
+
+    private void setBtnAddProfilePhoto(Button btnAddProfilePhoto) {
+        btnAddProfilePhoto.setOnClickListener(v -> {
+            onPickPhoto();
+        });
+    }
+
+
+    // Trigger gallery selection for a photo
+    public void onPickPhoto() {
+        // Create intent for picking a photo from the gallery
+        Intent intent = new Intent(Intent.ACTION_PICK,
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+
+        // If you call startActivityForResult() using an intent that no app can handle, your app will crash.
+        // So as long as the result is not null, it's safe to use the intent.
+        if (intent.resolveActivity(getPackageManager()) != null) {
+            // Bring up gallery to select a photo
+            startActivityForResult(intent, PICK_PHOTO_CODE);
+            PERSONALIZED_PHOTO_PICKED = true;
```

Comment:
I'm not sure if we still need this variable, but if we do we should set it in the "OnActivityResult" method. Right now we could return an error from the Activity but we'd still set this true regardless.